### DISCUSSION TITLE
Metadata cache coherence?

### DISCUSSION THREAD

**Discussion by eliphatfs** [seaweedfs/seaweedfs#5444] [Q&A]
It seems possible to happen like follows:
1) a file `A/B` is on seaweedfs.
2) Client 1 deleted file `A/B` syscall returns.
3) Client 2 code `ls A` shows `B`.
4) Metadata follow event handled.
5) Client 2 code read `A/B` ENOENT.

It that correct? If yes, is there anything we can do to improve? SeaweedFS changes claims to be visible on other clients upon file close but that is only for raw volume 'files' and not FS filer 'files'?

**Comment by n9** [comment]
Doesn't it depend on filer backend? (And filer's usage?)

**Reply by eliphatfs** [reply]
default filer setup without filer.toml, and backend is default leveldb2.
the metadata client-side cache and following logic is in seaweed.
